docs/architecture/adr-069-flexible-node-intitalization.md

```diff
@@ -0,0 +1,115 @@
+because these interfaces were _already_ changing as a result of the `p2p`
+[refactor](./adr-061-p2p-refactor-scope.md), so it made sense to think a bit
+more about how tendermint exposes this API.
+
+It is not immediately intuitive that the nexus of this change is in the
+relationship between the `rpc` package and the `node` package, particularly
+with regards to how the node instances are started. However, this is indeed
+the case and the existing dependencies are tangled around supporting these
+cases.
+
+## Observations
+
+To prepare for this work I've been playing in [this
+branch](https://github.com/tendermint/tendermint/tree/tychoish/scratch-node-minimize),
+with the node implementation to see if, without making major
+modifications or changing interfaces, what aspects could be
+simplified. My conclusions, generally: 
+
+- the implementation of the `node` service, doesn't really need the
+  ability to introspect the implementations of reactors in most
+  cases.
+
+- the exception to this, is the cluster of
+  consensus/statesync/blockhain (which depend on eachother,) because
+  `node` itself contains some of their specific initialization logic.
+
+- while the RPC environment construction happens in `node`, this can
+  happen early during `node` construction rather than at
+  initialization time, which reduces the need for the implementation
+  of the node object to track this state.
+
+Ideally, it seems like a node instance should be expressable simply as
+a list of `service.Services` and any knowledge of the internals of any
+of these services could be limited to their constructors. I sketched
+out a generic and simple `groupService` implementation, to handle a
+desecrate stage or sequence of services.  As some services have
+dependencies during construction or initialization and as a starting
+point, a cluster of dependencies can be wrapped as a single
+`service.Service`.
+
+## Goals
+
+- Provide a more flexible internal framework for initializing tendermint
+  nodes to make the initatilization process less hard-coded by the
+  implementation of the node objects. 
+  
+  - Reactors should not need to expose their interfaces *within* the
+    implementation of the node type, except in the context of some groups of
+    service. 
+
+- Expose some "unsafe"  way of replacing (some?) existing components, and or
+  inserting arbitrary reactors or services into the tendermint node's
+  initialization.
+  
+  - This may include creating an exported, top-level "unsafe" package with
+    several interfaces and constructors, or simply a function in the
+    node package. 
+
+- Refactor and simplify the process start/initialization logic,
+  separately from--but as a precursor to--permitting user replacement
+  of components.
+
+## Non-Goals
+
+- Fully abstract dependency-based process initialization. 
+
+- Fully abstract replacement of core tendermint components without vendoring.
+
+- Parallelized initialization of components.
```

Comment:
I don't know what any of these mean tbh. Also, there isn't a need for a newline between points.

We need a markdown linter @marbar3778 :)

Reply:
I think it improves readability, particularly with nested bulleted lists.

Also there is a markdown linter in the superlinter, and this seems to pass.

Reply:
I love it when people include non-goals in design docs! But like Bez, I'm not really sure what these mean. I have some guesses, and I _think_ I agree with them (esp. #2), but it would be best if you just elaborate/clarify instead.